test/unit/lib/plugins/aws/package/compile/events/cloud-front.test.js

```diff
@@ -1573,88 +1270,279 @@ describe('test/unit/lib/plugins/aws/package/compile/events/cloudFront.test.js',
+          fixture: 'function',
+          command: 'package',
+          configExt: {
+            functions: {
+              first: {
+                name: 'first',
+                handler: 'first.handler',
+                events: [
+                  {
+                    cloudFront: {
+                      eventType: 'viewer-request',
+                      origin: 's3://bucketname.s3.amazonaws.com/files',
+                      isDefaultOrigin: true,
+                    },
+                  },
+                ],
+              },
+              second: {
+                name: 'second',
+                handler: 'second.handler',
+                events: [
+                  {
+                    cloudFront: {
+                      eventType: 'viewer-request',
+                      origin: 's3://anotherbucket.s3.amazonaws.com/files',
+                      isDefaultOrigin: true,
+                    },
+                  },
+                ],
+              },
+            },
+          },
+        })
+      ).to.eventually.be.rejected.and.have.property(
+        'code',
+        'CLOUDFRONT_MULTIPLE_DEFAULT_ORIGIN_EVENTS'
+      );
     });
-
-    it.skip('TODO: should throw if none of the cloudfront events with different origins were defined as a default', async () => {
-      // Replaces
-      // https://github.com/serverless/serverless/blob/85e480b5771d5deeb45ae5eb586723c26cf61a90/lib/plugins/aws/package/compile/events/cloudFront/index.test.js#L1308-L1369
-
+    it('should throw if none of the cloudfront events with different origins were defined as a default', async () => {
       return expect(
-        runServerless({ fixture: 'function', command: 'package' })
-      ).to.eventually.be.rejected.and.have.property('code', 'TODO');
+        runServerless({
+          fixture: 'function',
+          command: 'package',
+          configExt: {
+            functions: {
+              first: {
+                name: 'first',
+                handler: 'first.handler',
+                events: [
+                  {
+                    cloudFront: {
+                      eventType: 'viewer-request',
+                      origin: 's3://bucketname.s3.amazonaws.com/files',
+                      pathPattern: '/files/*',
+                    },
+                  },
+                ],
+              },
+              second: {
+                name: 'second',
+                handler: 'second.handler',
+                events: [
+                  {
+                    cloudFront: {
+                      eventType: 'viewer-request',
+                      origin: 's3://anotherbucket.s3.amazonaws.com/files',
+                      pathPattern: '/anotherfiles/*',
+                    },
+                  },
+                ],
+              },
+            },
+          },
+        })
+      ).to.eventually.be.rejected.and.have.property(
+        'code',
+        'CLOUDFRONT_MULTIPLE_DEFAULT_ORIGIN_EVENTS'
+      );
```

Comment:
we are not using expect in `Should not throw if lambda config includes AllowedMethods or CachedMethods behaviour values and cache policy reference`. how that test case is helping us and what are we trying to achieve there

Reply:
@jindalAnuj that's a good point. I think this test should be reformulated to general test that confirms on most of the `cloud-front` event options, and confirms with `expect` that they landed at expected slots in CF template.

And that's actually covered in `Resource generation` section. So question do we have already this case over there. If so, then this test can be removed, and if not, question is can it be integrated over there, as that would seem as right direction

Reply:
@medikoo i have checked code it is a unique one but does not have a expect statements we should have it actually in resource validation IMO.

Reply:
Ok, so if it can be added to this singular  `runServerless` run as made in `Resource generation`, let's configure it there

Reply:
@medikoo i am removing it from here , we have coverage for it in 
`should support behavior customization` in `Resource generation`